akka-distributed-data/src/main/scala/akka/cluster/ddata/LogPayloadSize.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.ddata
+
+import akka.annotation.InternalApi
+import akka.cluster.ddata.Key.KeyId
+import akka.event.LoggingAdapter
+
+/**
+ * INTERNAL API
+ *
+ * This class is not thread-safe. It is supposed to be used from an actor.
+ */
+@InternalApi private[akka] class LogPayloadSize(log: LoggingAdapter, logSizeExceeding: Int, warnSizeExceeding: Int) {
+  private var maxPayloadBytes: Map[String, Int] = Map.empty
+
+  def logPayloadBytes(key: KeyId, size: Int): Unit = {
+    if (size >= logSizeExceeding) {
+      // 10% threshold until next log
+      def newMax = (size * 1.1).toInt
+
+      maxPayloadBytes.get(key) match {
+        case Some(max) =>
+          if (size > max) {
+            maxPayloadBytes = maxPayloadBytes.updated(key, newMax)
+            if (size >= warnSizeExceeding)
+              log.warning(
+                "New distributed data size for [{}] is [{}] bytes. Close to max remote message payload size.",
```

Comment:
I find the log message a bit misleading, it sounds like something is being reconfigured, what about: "New max distributed data size [{}] bytes seen for [{}]" or something along those lines?

Reply:
It's the word 'new' that is causing this confusion? Maybe we should find something.

Reply:
I have updated the log message, and also using same log message for the first and subsequent logs